src/order/filter/basic.lean

```diff
@@ -1446,6 +1446,69 @@ lemma tendsto_at_top [preorder β] (m : α → β) (f : filter α) :
   tendsto m f at_top ↔ (∀b, {a | b ≤ m a} ∈ f) :=
 by simp only [at_top, tendsto_infi, tendsto_principal]; refl
 
+lemma tendsto_at_top_mono' [preorder β] (l : filter α) ⦃f₁ f₂ : α → β⦄ (h : {x | f₁ x ≤ f₂ x} ∈ l) :
+  tendsto f₁ l at_top → tendsto f₂ l at_top :=
+assume h₁, (tendsto_at_top _ _).2 $ λ b, mp_sets ((tendsto_at_top _ _).1 h₁ b)
+  (monotone_mem_sets (λ a ha ha₁, le_trans ha₁ ha) h)
+
+lemma tendsto_at_top_mono [preorder β] (l : filter α) :
+  monotone (λ f : α → β, tendsto f l at_top) :=
+λ f₁ f₂ h, tendsto_at_top_mono' l $ univ_mem_sets' h
+
+section ordered_monoid
+
+variables [ordered_cancel_comm_monoid β] (l : filter α) {f g : α → β}
+
+lemma tendsto_at_top_add_nonneg_left' (hf : {x | 0 ≤ f x} ∈ l) (hg : tendsto g l at_top) :
```

Comment:
Why not even
```lean
lemma tendsto_at_top_add_left_of_le' (C : β) (hf : {x | C ≤ f x} ∈ l) (hg : tendsto g l at_top) :
  tendsto (λ x, f x + g x) l at_top
```

Reply:
I can prove this in an ordered group but not in an `ordered_cancel_comm_monoid`. If we know that `C ≤ f x` for all x and we need to show that `{x | C' ≤ f x + g x} ∈ l`, then which inequality of the form `?? ≤ g x` implies `C' ≤ f x + g x`?

Reply:
You are absolutely right, this is not true in general in an `ordered_cancel_comm_monoid` (with `-ℕ` and `f = -1` and `g = 0` as a counterexample). Could you please add it in an `ordered_group`, this will certainly be useful.

Reply:
Done.